python/lsst/ts/standardscripts/maintel/closed_loop.py

```diff
@@ -0,0 +1,519 @@
+# This file is part of ts_externalcripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["ClosedLoop"]
+
+import abc
+import enum
+import types
+import typing
+
+import numpy as np
+import yaml
+from lsst.ts import salobj
+from lsst.ts.observatory.control import RemoteGroup
+
+# TO-DO: Remove this import once the new LSSTCam is available
+from lsst.ts.observatory.control.maintel.comcam import ComCam, ComCamUsages
+from lsst.ts.observatory.control.maintel.mtcs import MTCS, MTCSUsages
+from lsst.ts.observatory.control.remote_group import Usages
+
+STD_TIMEOUT = 10
+
+
+# TODO - Subsitute by class in idl.enum when available
+class ClosedLoopMode(enum.IntEnum):
+    CWFS = 0
+    FAM = 1
+
+
+class DOFComponent(enum.IntEnum):
+    M2_dz = 0
+    M2_dx = 1
+    M2_dy = 2
+    M2_rx = 3
+    M2_ry = 4
+    Cam_dz = 5
+    Cam_dx = 6
+    Cam_dy = 7
+    Cam_rx = 8
+    Cam_ry = 9
+    M1M3_B1 = 10
+    M1M3_B2 = 11
+    M1M3_B3 = 12
+    M1M3_B4 = 13
+    M1M3_B5 = 14
+    M1M3_B6 = 15
+    M1M3_B7 = 16
+    M1M3_B8 = 17
+    M1M3_B9 = 18
+    M1M3_B10 = 19
+    M1M3_B11 = 20
+    M1M3_B12 = 21
+    M1M3_B13 = 22
+    M1M3_B14 = 23
+    M1M3_B15 = 24
+    M1M3_B16 = 25
+    M1M3_B17 = 26
+    M1M3_B18 = 27
+    M1M3_B19 = 28
+    M1M3_B20 = 29
+    M2_B1 = 30
+    M2_B2 = 31
+    M2_B3 = 32
+    M2_B4 = 33
+    M2_B5 = 34
+    M2_B6 = 35
+    M2_B7 = 36
+    M2_B8 = 37
+    M2_B9 = 38
+    M2_B10 = 39
+    M2_B11 = 40
+    M2_B12 = 41
+    M2_B13 = 42
+    M2_B14 = 43
+    M2_B15 = 44
+    M2_B16 = 45
+    M2_B17 = 46
+    M2_B18 = 47
+    M2_B19 = 48
+    M2_B20 = 49
+
+
+class ClosedLoop(salobj.BaseScript, metaclass=abc.ABCMeta):
```

Comment:
Not sure this is an appropriate name for the script. I think usually we want to reflect an action performed by the control system. Maybe drop the "d" and call it `CloseLoop`? However, I think the intention here is to create a base script to perform close loop with both ComCam and LSSTCam. If this is the case, I'd call this `BaseCloseLoop`.

I see that you prepared the class to do it (by adding the `metaclass`), however there's nothing in the implementation on that regard. I think you want to split this one in 2 scripts; a base and a ComCam implementation, leaving the LSSTCam implementation for later. Or maybe even doing the LSSTCam already after we merge the PR in ts_observatory_control.